features/networking/pod.feature

```diff
@@ -298,3 +298,89 @@ Feature: Pod related networking scenarios
+    Then the step should succeed
+    Given a pod becomes ready with labels:
+      | name=udp-pods |
+    And evaluation of `pod` is stored in the :host_pod1 clipboard
+
+    #Using node port to expose the service on port 8080 on the node IP address
+    When I run the :expose client command with:
+      | resource      | pod                      |
+      | resource_name | <%= cb.host_pod1.name %> |
+      | type          | NodePort                 |
+      | port          | 8080                     |
+      | protocol      | UDP                      |
+    Then the step should succeed
+    #Getting nodeport value
+    When I run the :get client command with:
+      | resource | service                                     |
+      | output   | jsonpath={.items[*].spec.ports[*].nodePort} |
+    Then the step should succeed
+    And evaluation of `@result[:response]` is stored in the :nodeport clipboard
+    
+    #Creating a simple client pod to generate traffic from it towards the exposed node IP address
+    When I run the :create client command with:
+      | f | https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/aosqe-pod-for-ping.json |
+    Then the step should succeed
+    Given a pod becomes ready with labels:
+      | name=hello-pod |
+    And evaluation of `pod` is stored in the :client_pod clipboard
+    # 'yes' command will send a character "h" continously for 3 seconds to /dev/udp on listener where the node is listening for udp traffic on exposed nodeport. The 3 seconds mechanism will create an Assured
+    #  entry which will give us enough time to validate upcoming steps
+    When I run the :exec background client command with:
+      | pod              | <%= cb.client_pod.name %>                             |
+      | oc_opts_end      |                                                       |
+      | exec_command     | bash                                                  |
+      | exec_command_arg | -c                                                    |
+      | exec_command_arg | yes "h">/dev/udp/<%= cb.node_ip %>/<%= cb.nodeport %> |
+    Given 3 seconds have passed
+    And I terminate last background process
+    
+    #Creating network test pod to levearage conntrack tool
+    When I run oc create over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/net_admin_cap_pod.yaml" replacing paths:
+      | ["spec"]["nodeName"] | <%= cb.nodes[0].name %> |
+    Then the step should succeed
+    Given a pod becomes ready with labels:
+      | name=network-pod |
+    And evaluation of `pod.name` is stored in the :network_pod clipboard
+    And I execute on the pod:
+      | bash | -c | conntrack -L \| grep "<%= cb.nodeport %>" |
+    Then the step should succeed
+    And the output should contain:
+      |<%= cb.host_pod1.ip %>|
+
+    #Deleting the udp listener pod which will trigger a new udp listener pod with new IP
+    When I run the :delete client command with:
```

Comment:
@anuragthehatter you should substitute the `:delete` call with `I ensure...`

Reply:
@pruan-rht Yep. I have used that most of the time. oops how can i forget that.

Reply:
Fixed and re-tested http://pastebin.test.redhat.com/843972